### PR TITLE
[DT-947]Fix flaky CohortEditor unit tests

### PR DESCRIPTION
**Pull request by rjohanek**
## Summary of changes:

Question: we haven't had any problems with flaky tests on DatasetBuilderDetails test or SpendReport test but they also both use the Chart component. Do we want to mock the Chart in those tests too and modify the tests that check for chart information?

### What
- Mock Chart component in CohortEditor test
- Unskip flaky  CohortEditor test
- Simplify flaky  CohortEditor test. The test case removed in this PR is already tested in a separate test (line 427). This just simplifies our longest unit test to test only one situation. 
- Removes Chart as the default export

### Why
There have been many test runs where the build fails because of failing cohort editor unit tests fail. For example, in[ this run,](https://app.circleci.com/pipelines/github/DataBiosphere/terra-ui/26361/workflows/0776de11-d3a4-406b-bc4c-b8531ce238d1/jobs/85777) two tests fail. The one to add/delete criteria groups and maintain expected names fails with a timeout, and the one to save a cohort fails with a console error. The save cohort test failure may be a side effect of the other failing test because I didn't see any runs where it failed by itself and it hasn't failed since the other test is being skipped. The failures only started after the charts were added to the CohortEditor page, so mocking the charts will likely help. 

### Testing strategy
After mocking the charts and simplifying the cohort editor test the runtime of the tests locally on my computer in down from ~13-15 seconds to ~4-5 seconds with the delete criteria group test going from ~3-5 seconds to ~0.5-1 second. This will probably decrease the likelihood of a timeout error.

### CODE REVIEW

File: src/components/Chart.js [hunk 1]
```diff
@@ -10,6 +10,4 @@ highchartsExporting(Highcharts);
 // this is needed to permit use of internally constructed links in a Highcharts formatter
 Highcharts.AST.bypassHTMLFiltering = true;
 
-const Chart = ({ options }) => h(HighchartsReact, { highcharts: Highcharts, options });
-
-export default Chart;
+export const Chart = ({ options }) => h(HighchartsReact, { highcharts: Highcharts, options });
```
Comment (rjohanek) [inline src/components/Chart.js:13]: removed default in order to make this component easier to test

File: src/dataset-builder/CohortEditor.test.ts [hunk 2]
```diff
@@ -436,25 +444,16 @@ describe('CohortEditor', () => {
...
     await user.click(screen.getByText('Add group'));
     // Assert
     expect(screen.getByText('Group 1')).toBeTruthy();
     expect(screen.getByText('Group 2')).toBeTruthy();
-    // Act
-    await user.click(screen.getByText('Add group'));
-    // Assert
     expect(screen.getByText('Group 3')).toBeTruthy();
     // Act
     await user.click(screen.getByLabelText('delete group 2'));
```
Comment (davidangb) [inline src/dataset-builder/CohortEditor.test.ts]: nit: in this test, is it worth adding some `toBeFalsy()` assertions? If the page loads with "Group 1", "Group 2", and "Group 3" text already present on the page, the test will pass regardless of which clicks it emulates.

I realize this is out of scope for this ticket, take this feedback as optional

**Review by davidangb — APPROVED** [review]
one optional comment inline. I didn't try it myself, but code looks good. Thank you!

File: src/dataset-builder/CohortEditor.test.ts (hunk 2 above)
Comment (rjohanek) [reply src/dataset-builder/CohortEditor.test.ts]: good idea!

File: src/dataset-builder/CohortEditor.test.ts [hunk 3]
```diff
@@ -436,31 +444,23 @@ describe('CohortEditor', () => {
...
     showCohortEditor();
     const user = userEvent.setup();
-    // Assert
-    expect(screen.getByText('Group 1')).toBeTruthy();
     // Act
     await user.click(screen.getByText('Add group'));
-    // Assert
-    expect(screen.getByText('Group 2')).toBeTruthy();
-    // Act
-    await user.click(screen.getByLabelText('delete group 2'));
```
Comment (pshapiro4broad) [inline src/dataset-builder/CohortEditor.test.ts]: This is removing a test case that isn't tested elsewhere. The test here is that, after a group is deleted, the group numbers shown to the user use the ordering of groups on screen. Before it was fixed, this deletion would result in a "skipped" number as the deleted group number wouldn't be re-used for the next created group.

File: src/dataset-builder/CohortEditor.test.ts [hunk 4]
```diff
@@ -436,31 +444,23 @@ describe('CohortEditor', () => {
...
     await user.click(screen.getByText('Add group'));
```
Comment (pshapiro4broad) [inline src/dataset-builder/CohortEditor.test.ts]: If you change the test to no longer delete group 2, then this is adding group 3, which isn't needed until after the assert on line 456.

File: src/dataset-builder/CohortEditor.test.ts (hunk 2 above)
Comment (pshapiro4broad) [reply src/dataset-builder/CohortEditor.test.ts]: > is it worth adding some toBeFalsy() assertions? If the page loads with "Group 1", "Group 2", and "Group 3" text already present on the page, the test will pass regardless of which clicks it emulates

That's true, and there's no harm in doing this, but I don't think we do this in other cases. Personally I would test for cases we know can (or have) happened. If this problem ever comes up, and we need to fix it, we could add a test for it then.

**Review by pshapiro4broad — APPROVED** [review]
Looks OK but I would revert the changes to the CohortEditor test to restore its previous functionality.

If the Chart code is known to be slow, is there a way to change the render step in CohortEditor to render a different component that doesn't include the chart? Maybe use `CohortEditorContents` for tests that don't need the `Save cohort` button?

File: src/dataset-builder/CohortEditor.test.ts (hunk 3 above)
Comment (rjohanek) [reply src/dataset-builder/CohortEditor.test.ts]: I believe that is tested by the test above this one with the description "produces user understandable criteria group names after deletion"

File: src/dataset-builder/CohortEditor.test.ts (hunk 4 above)
Comment (rjohanek) [reply src/dataset-builder/CohortEditor.test.ts]: the deletion of group 2 happens below on line 259, then the test confirms that group 3 is now called group 2

**Comment by rjohanek** [comment]
> If the Chart code is known to be slow, is there a way to change the render step in CohortEditor to render a different component that doesn't include the chart? Maybe use `CohortEditorContents` for tests that don't need the `Save cohort` button?

We could render just the contents for some tests. We would still need to mock the chart for the other tests such as the save cohort tests. So, this suggestion could speed up some tests, but wouldn't remove the need for mocking this component. Maybe, we could see if this current approach is fast enough, if we still run into problems with timeouts, then I think we could try this suggestion. What do you think?

File: src/dataset-builder/CohortEditor.test.ts (hunk 3 above)
Comment (pshapiro4broad) [reply src/dataset-builder/CohortEditor.test.ts]: OK, I see that the above test is intending to test this case. Maybe it should be merged into this case completely?

Also right now that test is a bit confusing in what it chooses to assert or not. After the first `Add group`, there's both `Group 1` and `Group 2`. It then deletes group 1, at which point there's only `Group 1` left (since what was 2 before is now 1). It then adds another group, and asserts that `Group 1` is visible, but both 1 and 2 should be visible now. I think this odd construction may be because when it was originally written, the cohort editor initially had no groups, so in that test case the test is only expecting to see `Group 1` and not expecting to see `Group 2` at all.

Comment (cinyecai) [reply src/dataset-builder/CohortEditor.test.ts]: I think Phil's right, these tests were written before the cohort editor automatically rendered the first group. On that point, the "produces user understandable criteria group names after deletion" was intended to test that when deleting all groups and adding a new one, it starts at "Group 1" and not at whatever number of groups came before it

**Comment by pshapiro4broad** [comment]
> Maybe, we could see if this current approach is fast enough, if we still run into problems with timeouts, then I think we could try this suggestion. What do you think?

I think it's ok to leave it as-is and revisit it when we go back and add support for chart data that will need testing. Then it might make sense to split out the graph rendering tests from the non-ones.

File: src/dataset-builder/CohortEditor.test.ts (hunk 3 above)
Comment (rjohanek) [reply src/dataset-builder/CohortEditor.test.ts]: Okay that makes sense I'll just revert this change
